stacker/plan.py

```diff
@@ -210,65 +220,40 @@ def _single_run(self):
                 )
                 continue
 
-            # Kick off watchers - used for tailing the stack
-            if (
-                not step.done and
-                self._watch_func and
-                step_name not in self._watchers
-            ):
-                process = multiprocessing.Process(
-                    target=self._watch_func,
-                    args=(step.stack,)
-                )
-                self._watchers[step_name] = process
-                process.start()
-
-            try:
-                status = step.run()
-            except CancelExecution:
-                status = SkippedStatus(reason="canceled execution")
-
-            if not isinstance(status, Status):
-                raise ValueError(
-                    "Step run_func must return a valid Status object. "
-                    "(Returned type: %s)" % (type(status)))
-            step.set_status(status)
+            if not step.submitted:
+                try:
+                    status = step.run()
+                except CancelExecution:
+                    status = SkippedStatus(reason="canceled execution")
 
-            # Terminate any watchers when step completes
-            if step.done and step_name in self._watchers:
-                self._terminate_watcher(self._watchers[step_name])
+                if not isinstance(status, Status):
```

Comment:
I wonder if this should just be moved into `Step.set_status`, what do you think?